plugins/modules/lvg.py

```diff
@@ -255,161 +287,80 @@ def find_vg_state(module, vg_name):
-    rc, active_vgs, stderr = module.run_command(cmd)
-    if rc != 0:
-        result['msg'] = "Command '%s' failed." % cmd
-        result['cmd'] = cmd
-        result['rc'] = rc
-        result['stdout'] = active_vgs
-        result['stderr'] = stderr
-        module.fail_json(**result)
+    fail_msg = "Command '%s' failed." % cmd
+    run_cmd(module, cmd, None, fail_msg, fetch=True)
+    active_vgs = result['stdout']
 
-    rc, all_vgs, stderr = module.run_command("lsvg")
-    if rc != 0:
-        result['msg'] = "Command 'lsvg' failed."
-        result['cmd'] = cmd
-        result['rc'] = rc
-        result['stdout'] = all_vgs
-        result['stderr'] = stderr
-        module.fail_json(**result)
+    cmd = 'lsvg'
+    fail_msg = "Command '%s' failed." % cmd
+    run_cmd(module, cmd, None, fail_msg, fetch=True)
+    all_vgs = result['stdout']
 
     if vg_name in all_vgs and vg_name not in active_vgs:
         return False
-
     if vg_name in active_vgs:
         return True
-
     return None
 
 
-def change_vg(module, vg_name, vg_state):
+def change_vg(module, vg_name, init_props):
     """
-    Modifies/Extends volume group
+    Modifies volume group
     arguments:
         module: Ansible module argument spec.
         vg_name: Volume group name
-        vg_state: Volume group state
+        init_props: Initial properties of the volume group
     note:
         Exits with fail_json in case of error
     return:
         none
     """
     global result
-    pvs = module.params['pvs']
-    force = module.params['force']
-    fopt = ""
-    if force:
-        fopt = "-f "
-
-    if module.params["major_num"] or module.params["pp_size"] or module.params["mirror_pool"]:
-        result['msg'] = "Attributes major_num, pp_size and mirror_pool are not supported while changing volume group %s." % vg_name
-        module.fail_json(**result)
 
-    # if pvs, extend the volume group
-    # if other attributes specified, change that too
-
-    if pvs:
-        # extend VG
-        cmd = "extendvg %s %s %s" % (fopt, vg_name, ' '.join(pvs))
-        rc, stdout, stderr = module.run_command(cmd)
-        if rc != 0:
-            result['msg'] = "Failed to extend volume group %s. Command '%s' failed.\n" % (vg_name, cmd)
-            result['cmd'] = cmd
-            result['rc'] = rc
-            result['stdout'] = stdout
-            result['stderr'] = stderr
-            module.fail_json(**result)
+    major_num = module.params["major_num"]
+    pp_size = module.params["pp_size"]
+    mirror_pool = module.params["mirror_pool"]
+    if major_num or pp_size or mirror_pool:
+        result['msg'] += "Attributes major_num, pp_size or mirror_pool "
+        result['msg'] += "are not supported while changing volume group %s.\n" % vg_name
+
+    # get initial vg type
+    pattern = r"^(MAX PPs per VG:)\s+(\d+)"
+    max_pp_per_vg = re.search(pattern, init_props, re.MULTILINE)
+    max_pp_per_vg = int(max_pp_per_vg.groups()[1])
+    if max_pp_per_vg == 32512:
```

Comment:
Can we define the variables for MAX PP size for small,big and scalable VG here?

Reply:
Ah yes, that would look cleaner, i will be doing that. define it as constants in this module